frontend/packages/sdk/src/pruntime/createPruntimeClient.ts

```diff
@@ -0,0 +1,37 @@
+import { fetch } from 'undici'
+import { prpc, pruntime_rpc as pruntimeRpc } from './proto'
+
+/**
+ * Create a http client prepared for protobuf
+ *
+ * @param baseURL The base URL of the pruntime server
+ * @returns A PhactoryAPI object
+ */
+export default function createPruntimeClient(baseURL: string) {
+  const pruntimeApi = pruntimeRpc.PhactoryAPI.create(async (method, requestData, callback) => {
+    try {
+      const resp = await fetch(`${baseURL}/prpc/PhactoryAPI.${method.name}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/octet-stream',
+        },
+        body: new Uint8Array(requestData),
+      })
+      if (resp.status === 200) {
+        const buffer = await (await resp.blob()).arrayBuffer()
+        callback(null, new Uint8Array(buffer))
+      } else if (resp.status === 500) {
+        const buffer = await (await resp.blob()).arrayBuffer()
+        const prpcError = prpc.PrpcError.decode(new Uint8Array(buffer))
```

Comment:
If the error come from Nginx, it may not be a protobuf object. Consider to handle plain text case?